<?xml version="1.0"?>
<sdf version="1.4">
  <world name="default">
    <!-- <include filename="ground_plane.model"/>
         <include filename="sun.light"/> -->
    <include>
      <uri>model://ground_plane</uri>
    </include>
    
    <include>
      <uri>model://sun</uri>
    </include>
    
    <model name="box1">
      <pose>0 0 0.5 0 0 0</pose>
      <link name="link">
        <collision name="collision">
          <geometry>
            <box>
              <size>1 1 1</size>
            </box>
          </geometry>
        </collision>
        
        <visual name="visual">
          <geometry>
            <box>
              <size>1 1 1</size>
            </box>
          </geometry>
        </visual>
      </link>
      <!--plugin name="usbl_transceiver" filename="libtransceiverPlugin.so">
        <namespace>USBL</namespace>
        <transponder_device>transponder_manufacturer</transponder_device>
        <transponder_ID>1</transponder_ID>
        <transceiver_device>transceiver_manufacturer</transceiver_device>
        <transceiver_ID>168</transceiver_ID>
        <enable_ping_scheduler>false</enable_ping_scheduler>
        <transponder_attached_object>sphere</transponder_attached_object>
      </plugin-->
    </model>

    <model name="box2">
      <pose>5 5 0.5 0 0 0</pose>
      <link name="link">
        <collision name="collision">
          <geometry>
            <box>
              <size>1 1 1</size>
            </box>
          </geometry>
        </collision>
        
        <visual name="visual">
          <geometry>
            <box>
              <size>1 1 1</size>
            </box>
          </geometry>
        </visual>
      </link>
      <!--plugin name="usbl_transceiver" filename="libtransceiverPlugin.so">
        <namespace>USBL</namespace>
        <model_name>box2</model_name>
        <transponder_device>transponder_manufacturer</transponder_device>
        <transponder_ID>1</transponder_ID>
        <transceiver_device>transceiver_manufacturer</transceiver_device>
        <transceiver_ID>168</transceiver_ID>
        <enable_ping_scheduler>false</enable_ping_scheduler>
        <transponder_attached_object>sphere</transponder_attached_object>
      </plugin-->
    </model>
    
    <model name="sphere1">
      <pose>3 3 0.5 0 0 0</pose>
      <link name="link">
        <collision name="collision">
          <geometry>
            <sphere>
              <radius>0.7</radius>
            </sphere>
          </geometry>
        </collision>
	
        <visual name="visual">
          <geometry>
            <sphere>
              <radius>0.7</radius>
            </sphere>
          </geometry>
        </visual>
      </link>
      <!--plugin name="usbl_transponder" filename="libtransponderPlugin.so">
        <namespace>USBL</namespace>
        <transponder_device>transponder_manufacturer</transponder_device>
        <transponder_ID>1</transponder_ID>
        <transceiver_device>transceiver_manufacturer</transceiver_device>
        <transceiver_ID>168</transceiver_ID>
        <transceiver_model_name>square</transceiver_model_name>
        <mu>0</mu>
        <sigma>0.0</sigma>
      </plugin-->
    </model>

    <model name="sphere2">
      <pose>-3 3 0.5 0 0 0</pose>
      <link name="link">
        <collision name="collision">
          <geometry>
            <sphere>
              <radius>0.7</radius>
            </sphere>
          </geometry>
        </collision>
	
        <visual name="visual">
          <geometry>
            <sphere>
              <radius>0.7</radius>
            </sphere>
          </geometry>
        </visual>
      </link>
      <!--plugin name="usbl_transponder" filename="libtransponderPlugin.so">
        <namespace>USBL</namespace>
        <transponder_device>transponder_manufacturer</transponder_device>
        <transponder_ID>2</transponder_ID>
        <transceiver_device>transceiver_manufacturer</transceiver_device>
        <transceiver_ID>168</transceiver_ID>
        <transceiver_model_name>box</transceiver_model_name>
        <mu>0</mu>
        <sigma>0.0</sigma>
      </plugin-->
    </model>

    <!--model name="sphere3">
      <pose>-3 -3 0.5 0 0 0</pose>
      <link name="link">
        <collision name="collision">
          <geometry>
            <sphere>
              <radius>0.7</radius>
            </sphere>
          </geometry>
        </collision>
	
        <visual name="visual">
          <geometry>
            <sphere>
              <radius>0.7</radius>
            </sphere>
          </geometry>
        </visual>
      </link>
      <plugin name="usbl_transponder" filename="libtransponderPlugin.so">
        <namespace>USBL</namespace>
        <transponder_device>transponder_manufacturer</transponder_device>
        <transponder_ID>2</transponder_ID>
        <transceiver_device>transceiver_manufacturer</transceiver_device>
        <transceiver_ID>168</transceiver_ID>
        <model_name>"box1"</model_name>
        <mu>0</mu>
        <sigma>0.0</sigma>
      </plugin>
    </model-->
  </world>
</sdf>